game/scripts/vscripts/internal/util.lua

```diff
@@ -6,14 +15,88 @@ function split(s, delimiter)
     return result;
 end
 
+function regexsplit(s, delimiter)
+    result = {};
+    for match in s:gmatch("([^"..delimiter.."]+)") do
+        table.insert(result, match);
+    end
+    return result;
+end
+
+function TracesFromFilename (filename)
+  local traces = {}
+  local i = 1
+
+  local parts = regexsplit(filename, '%s/\\')
+  local partialTrade = nil
+  for i, part in ipairs(parts) do
+    if partialTrade == nil and part ~= "components" then
+      partialTrade = part
+      table.insert(traces, partialTrade .. ":*")
+    elseif partialTrade ~= nil then
+      partialTrade = partialTrade .. ":" .. part
+      table.insert(traces, partialTrade .. ":*")
+    end
+  end
+
+  table.insert(traces, partialTrade)
+
+  return traces
+end
+
+function IsAnyTraceEnabled (traces)
+  if Debug.EnableAll then
+    return true
+  end
+
+  for i, trace in ipairs(traces) do
+    if Debug.EnabledModules[trace] then
+      return true
+    end
+  end
+
+  return false
+end
+
+-- written by yeahbuddy, taken from https://github.com/AngelArenaAllstars/aaa/pull/80
+-- modified for clarity
+function GetCallingFile (offset)
+  offset = offset or 4
+
+  local str = debug.traceback()
+  local lines = split(str, '\n')
+  local line = lines[offset]
+  local dirName , lineNo= string.match(line, "scripts[/\\]vscripts[/\\](.+).lua:([0-9]+):")
```

Comment:
Also splitting on `/` and `\`.

`[/\\]` is a character group of / and \, so it'll grab either.